Fix when using Facebook Menu to log out in a FeedDialog #103 (Tip N°2)

AlanGusLive opened the pull request:
Fix when using Facebook Menu to log out in a FeedDialog (the www.facebook.com/logout.php is called but the token wasn't deleted).

msftclas commented:
Hi __@AlanGusLive__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

adiviness commented on samples/LoginCpp-UWP/LoginCpp/Dialogs.xaml.cpp:
```diff
@@ -137,16 +137,24 @@ void Dialogs::Feed_Click(
-        OutputDebugString(L"Showed 'Feed' dialog.\n");
-    });
+
+	if (s->AccessTokenData == nullptr)
```
This should use `s->LoggedIn` instead to check if the user is still logged in.

adiviness commented on samples/LoginCpp-UWP/LoginCpp/Dialogs.xaml.cpp:
```diff
@@ -156,14 +164,22 @@ void Dialogs::AppRequests_Click(
-    {
-        OutputDebugString(L"Showed 'Requests' dialog.\n");
-    });
+	if (s->AccessTokenData == nullptr)
```
This should use `s->LoggedIn` instead to check if the user is still logged in.

adiviness commented on samples/LoginCpp-UWP/LoginCpp/Dialogs.xaml.cpp:
```diff
@@ -137,16 +137,24 @@ void Dialogs::Feed_Click(
+
+	if (s->AccessTokenData == nullptr)
+	{
+		OutputDebugString(L"User is not logged (access token was deleted).\n");
```
We don't know for sure that the cause of the error is that the access token was deleted, the string should just mention that the user is no longer logged in.

adiviness commented on samples/LoginCpp-UWP/LoginCpp/Dialogs.xaml.cpp:
```diff
@@ -156,14 +164,22 @@ void Dialogs::AppRequests_Click(
-    });
+	if (s->AccessTokenData == nullptr)
+	{
+		OutputDebugString(L"User is not logged (access token was deleted).\n");
```
We don't know for sure that the cause of the error is that the access token was deleted, the string should just mention that the user is no longer logged in.

adiviness commented on winsdkfb/winsdkfb/winsdkfb.Shared/FacebookDialog.xaml.cpp:
```diff
@@ -463,6 +471,17 @@ void FacebookDialog::dialogWebView_FeedNavStarting(
+		FBSession^ sess = FBSession::ActiveSession;
+		sess->LogoutAsync();
+
+		FBError^ err = FBError::FromJson(ref new String(ErrorObjectJson));
```
`ErrorObjectJson` should not be used here because the error message would not accurately describe what happened in this case. A new json error string should be used instead with a new code (4202?) and a more accurate error message.

adiviness commented:
Thanks for changing the pull request to target the `develop` branch. I've marked a few lines with specific changes that should be made before I can accept the pull request. I also noticed a few other things that should be changed. 

It looks like you are using tabs for the lines added, they should be standardized to whatever the blocks around them are using (which I think is spaces for all of them) so that there isn't weird indentation when viewing the code with some programs.

The code added to `FacebookDialog::dialogWebView_FeedNavStarting` that calls `IsLogoutRedirect` should also be run when `FacebookDialog::dialogWebView_RequestNavStarting` and `FacebookDialog::dialogWebView_SendNavStarting` are called so that the sdk will behave correctly if the user logs out from within the request or send dialogs as well.

The checks that were added to `samples/LoginCpp-UWP/LoginCpp/Dialogs.xaml.cpp` that ensure that the user is still logged in before calling a dialog function should also be present in the non-UWP LoginCpp sample. It is very similar to LoginCpp-UWP except that the desktop and phone code is split up into two directories, so that sample actually has two `Dialogs.xaml.cpp` files that should be updated to match the UWP sample.

AlanGusLive commented:
Hi adiviness,

I have updated the pull request. With Visual Studio 2015, by default the tabs are not replaced by spaces, so I changed the configuration of VS2015 to replace tabs by spaces (Tools->Options->Text Editor->All Languages->Tabs->Insert spaces). I used the menu "Edit->Advanced->Format Document" to replace all the tabs by spaces in the files I have changes.

Regards,
AlanGusLive

adiviness commented:
Looks good, thanks for making the changes!
